clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -143,9 +143,12 @@ private boolean matchesRequestedPartitions(FetchRequest request, FetchResponse r
     /**
      * Set-up a fetch request for any node that we have assigned partitions for which doesn't already have
      * an in-flight fetch or pending fetch data.
+     * @return false if no fetches were sent, true otherwise
      */
-    public void sendFetches() {
+    public boolean sendFetches() {
+        boolean fetches = false;
         for (Map.Entry<Node, FetchRequest> fetchEntry : createFetchRequests().entrySet()) {
```

Comment:
While you're here, I suggest fixing `fetchablePartitions` so that it doesn't do two `remove` calls on an `ArrayList` (in the worst case, it has to shift all the elements in the underlying array twice). We could pass a predicate to `subscriptions.fetchablePartitions()` and make it more efficient.